Python/compile.c

```diff
@@ -7745,6 +7746,120 @@ assemble_jump_offsets(struct assembler *a, struct compiler *c)
     } while (extended_arg_recompile);
 }
 
+
+// Ensure each basicblock is only put onto the stack once.
+#define MAYBE_PUSH(B) do {                          \
+        if ((B)->b_visited == 0) {                  \
+            *(*stack_top)++ = (B);                  \
+            (B)->b_visited = 1;                     \
+        }                                           \
+    } while (0)
+
+static void
+scan_block_for_local(int target, basicblock *b, bool unsafe_to_start,
+                     basicblock ***stack_top)
+{
+    bool unsafe = unsafe_to_start;
+    for (int i = 0; i < b->b_iused; i++) {
+        struct instr *instr = &b->b_instr[i];
+        if (instr->i_oparg != target) {
+            continue;
+        }
+        if (unsafe && instr->i_except != NULL) {
+            MAYBE_PUSH(instr->i_except);
+        }
```

Comment:
Might be worth asserting that the opcode is not a super-instruction (in case the compiler code changes and this is moved too late).

Reply:
I added an assertion above that there are no `EXTENDED_ARG`s. Does that work?

Reply:
I meant an assertion that it's not LOAD_FAST__LOAD_FAST or one of those. Would that be correct?

Reply:
Oh, yes, added now.